uber/site_sections/registration.py

```diff
@@ -162,14 +163,20 @@ def qrcode_generator(self, data):
 
         Returns: A PNG buffer. Use this function in an img tag's src='' to display an image.
 
+        NOTE: this will be called directly by attendee's client browsers to display their QR code.
+        This will potentially be called on the order of 100,000 times per event and serve up a lot of data.
+        Be sure that any modifications to this code are fast and don't unnecessarily increase CPU load.
+
+        If you run into performance issues, consider using an external cache to cache the results of
+        this function.  Or, offload image generation to a dedicated microservice that replicates this functionality.
+
         """
-        if not c.QR_CODE_PASSWORD:
-            return False  # Don't generate a QR Code if we can't encrypt it
+        if not c.QR_CODE_PASSWORD or not qr_cipher:
+            # Don't generate a QR Code if we aren't configured for encryption
+            return 'QR code password not set, cannot generate images'
```

Comment:
Since this is being invoked by a ``<img>`` tag, the only way this would really be shown to end users is if we returned an actual png file which shows this text.  But that's probably overkill.